### ISSUE TITLE
Issue generating JavaDoc for member variables

### ISSUE THREAD

**Issue by srihari-sridharan** [madhavd1/vscode-javadoc-tools#1]
First, appreciations for your contribution to the community of developers.
Just a minor defect. Java doc is not getting associated with the member variables.

```
    private final AlertLevel alertLevel;
    private final String eventSource;
    private final Object payload;

    

/** 
 * @return AlertLevel
 */

/** 
 * @return String
 */

/** 
 * @return Object
 */
```
instead of
```
    /**
     * @return AlertLevel
     */
    private final AlertLevel alertLevel;

    /**
     * @return String
     */
    private final String eventSource;

    /**
     * @return Object
     */
    private final Object payload;
```

**Comment by cruizba** [comment]
How did you get that? I think the extension only works at the moment for methods.
Btw, @madhavd1 thanks for your contribution :)

**Comment by madhavd1** [comment]
Thanks for the appreciation guys!
@cruizba is right that the extension only generates javadoc for methods. Ideally it shouldn't create comments for anything else.
I'm still working on handling javadoc comments for other class symbols.

**Comment by srihari-sridharan** [comment]
I was using this in a spring project that uses lombok for getters and
setters.

**Comment by madhavd1** [comment]
@srihari-sridharan, coincidentally my project for testing out this extension is also a Spring boot project with Lombok.
I just tried again and was not able to replicate this issue.

Would it be possible for you to provide the file where you are able to replicate it?

**Comment by srihari-sridharan** [comment]
Hi Madhav,

Since the code is confidential, I wouldn't be able to share the file as-is.

I just had a simple class with 3 private final variables and a
parameterized constructor.



Thanks and Regards,

*Srihari Sridharan*

Never postpone work which should be done without delaying. Tirukkuṛaḷ
<http://en.wikipedia.org/wiki/Tirukkural> *2nd ** century BC*

தூங்குக தூங்கிச் செயற்பால் தூங்கற்க

தூங்காது செய்யும் வினை. - திருக்குறள்
<http://en.wikipedia.org/wiki/Tirukkural>

**Comment by madhavd1** [comment]
Hi @srihari-sridharan,
I'm still not able to replicate this.
The extension only picks up methods from all the DocumentSymbols by comparing with vscode.SymbolKind.Method.
It shouldn't be picking up member variables unless it is reading them as "methods" instead of "fields".

Can you please check if that's the case - In case your member variables are being falsely read as Methods, you will see the purple box next to them instead of blue.
![image](https://user-images.githubusercontent.com/30430070/64520468-def5c980-d313-11e9-81c2-90ed9ab22be1.png)

Madhav

**Comment by srihari-sridharan** [comment]
Ok will check. Thanks for the clarification.

**Comment by rz-itac** [comment]
hello, I can confirm the issue, my setup is as follows:

vscode: 1.48.2
javadoc tools: 1.4.0
gabrielbb.vscode-lombok: 1.0.1 (this extension is for lombok)
I also have extensions for java/spring, if you still cannot replicate I can share my extension list.

I have a maven project with lombok (no spring): and i have a class like this:

`
@Getter
@Setter
@ToString
@EqualsAndHashCode(of = {"fecha", "evento", "usuario", "ip"})
@NoArgsConstructor
public class DataDTO implements Serializable {

  private static final long serialVersionUID = -4892590542766570450L;

  private Date fecha;
  private Integer evento;
  private String usuario;
  private String ip;

  public DataDTO(Date fecha, Integer evento, String usuario, String ip) {
    this();
    this.fecha = fecha;
    this.evento = evento;
    this.usuario = usuario;
    this.ip = ip;
  }

  public void addValidacionFallida(String msg) {
    evento = null;
  }
}`

when I run "Generate javadoc for current file" I get:

`
/** 
 * @return Date
 */

/** 
 * @return Integer
 */

/** 
 * @return String
 */

/** 
 * @return String
 */
@Getter
@Setter

/** 
 * @return String
 */
@ToString

/** 
 * @return boolean
 */

/** 
 * @return boolean
 */

/** 
 * @return int
 */
@EqualsAndHashCode(of = {"fecha", "evento", "usuario", "ip"})
@NoArgsConstructor
public class DataDTO implements Serializable {

  private static final long serialVersionUID = -4892590542766570450L;

  private Date fecha;
  private Integer evento;
  private String usuario;
  private String ip;

  public DataDTO(Date fecha, Integer evento, String usuario, String ip) {
    this();
    this.fecha = fecha;
    this.evento = evento;
    this.usuario = usuario;
    this.ip = ip;
  }

  /**
   * @param msg
   */
  public void addValidacionFallida(String msg) {
    evento = null;
  }
}
`

If you see, above the @Getter and @Setter annotations, javadoc blocks are generated as well as for @ToString and hashcode.

Thanks for your extension.

**Comment by madhavd1** [comment]
Hi @rz-itac ,
Thanks for the details. I'm looking into this.

**Comment by madhavd1** [comment]
Hi @rz-itac, @srihari-sridharan 
I was able to replicate the issue now. 
I'm working on fixing the issue. Will let you know once the fix is available.

Madhav

**Comment by madhavd1** [comment]
Fixed with 1e508e146d4e05241e6f18d8e693620ef3f03cf0
